src/waveWorker.js

```diff
@@ -1,119 +1,124 @@
 
-    case 'record':
-      if ( wavPCM ) {
-        wavPCM.record( e['data']['buffers'] );
-      }
-      break;
+(function( global ) {
+  var encoder;
+  global['onmessage'] = function( e ){
+    switch( e['data']['command'] ){
 
-    case 'done':
-      if ( wavPCM ) {
-        wavPCM.requestData();
-      }
-      break;
-
-    case 'init':
-      wavPCM = new WavePCM( e['data'] );
-      break;
-
-    default:
-      // Ignore any unknown commands and continue recieving commands
-  }
-};
-
-var WavePCM = function( config ){
-  this.sampleRate = config['sampleRate'] || 48000;
-  this.bitDepth = config['bitDepth'] || 16;
-  this.recordedBuffers = [];
-  this.bytesPerSample = this.bitDepth / 8;
-};
-
-WavePCM.prototype.record = function( buffers ){
-  this.numberOfChannels = this.numberOfChannels || buffers.length;
-  var bufferLength = buffers[0].length;
-  var reducedData = new Uint8Array( bufferLength * this.numberOfChannels * this.bytesPerSample );
-
-  // Interleave
-  for ( var i = 0; i < bufferLength; i++ ) {
-    for ( var channel = 0; channel < this.numberOfChannels; channel++ ) {
-      
-      var outputIndex = ( i * this.numberOfChannels + channel ) * this.bytesPerSample;
-      var sample = buffers[ channel ][ i ];
-      
-      // Check for clipping
-      if ( sample > 1 ) {
-        sample = 1;
-      }
+      case 'encode':
+        if (encoder) {
+          encoder.record( e['data']['buffers'] );
+        }
+        break;
 
-      else if ( sample < -1 ) {
-        sample = -1;
-      }
+      case 'done':
+        if (encoder) {
+          encoder.requestData();
+        }
+        break;
+
+      case 'init':
+        encoder = new WavePCM( e['data'] );
+        break;
 
-      // bit reduce and convert to uInt
-      switch ( this.bytesPerSample ) {
-        case 4:
-          sample = sample * 2147483648;
-          reducedData[ outputIndex ] = sample;
-          reducedData[ outputIndex + 1 ] = sample >> 8;
-          reducedData[ outputIndex + 2 ] = sample >> 16;
-          reducedData[ outputIndex + 3 ] = sample >> 24;
-          break;
-
-        case 3:
-          sample = sample * 8388608;
-          reducedData[ outputIndex ] = sample;
-          reducedData[ outputIndex + 1 ] = sample >> 8;
-          reducedData[ outputIndex + 2 ] = sample >> 16;
-          break;
-
-        case 2:
-          sample = sample * 32768;
-          reducedData[ outputIndex ] = sample;
-          reducedData[ outputIndex + 1 ] = sample >> 8;
-          break;
-
-        case 1:
-          reducedData[ outputIndex ] = ( sample + 1 ) * 128;
-          break;
-
-        default:
-          throw "Only 8, 16, 24 and 32 bits per sample are supported";
+      default:
+        // Ignore any unknown commands and continue recieving commands
+    }
+  };
+
+  var WavePCM = function( config ){
+    this.sampleRate = config['encoderSampleRate'] || 44100;
```

Comment:
I feel that using `encoder...` in the config naming is incorrect, as in a riff WAV file there is no encoding going on.       Perhaps we could use `wavSampleRate` and `wavBitDepth` and specify in the readme that these are the only config options that work with wav.